core/src/test/scala/org/apache/spark/scheduler/DAGSchedulerSuite.scala

```diff
@@ -143,7 +143,7 @@ class DAGSchedulerSuite extends SparkFunSuite with LocalSparkContext with TimeLi
         execId: String,
         accumUpdates: Array[(Long, Seq[AccumulatorV2[_, _]])],
         blockManagerId: BlockManagerId,
-        executorUpdates: ExecutorMetrics): Boolean = true
+        executorUpdates: scala.collection.Map[(Int, Int), ExecutorMetrics]): Boolean = true
```

Comment:
Your question goes to the signature in TaskScheduler#executorHeartbeatReceived then. I guess I was used to using java.util.Map and java.util.List in java-land, so I thought the more general type is preferred in the signature. I used a scala.collection.Map for executorUpdates in the case class Heartbeat, and so that percolated through.
Does it make a difference? It is true that I actually pass a mutable.Map in the Heartbeat.

Reply:
I changed executorUpdates to be a scala.collection.mutable.Map in Heartbeat and in the methods in classes that receive it from the heartbeat.